pageserver/src/consumption_metrics.rs

```diff
@@ -143,6 +145,12 @@ async fn collect_metrics(
         // these are point in time, with variable "now"
         let metrics = metrics::collect_all_metrics(&tenant_manager, &cached_metrics, &ctx).await;
 
+        // Pre-generate event idempotency keys, to reuse them across the bucket
+        // and HTTP sinks.
+        let idempotency_keys = std::iter::repeat_with(|| node_id.as_str().generate())
+            .take(metrics.len())
+            .collect_vec();
```

Comment:
Doesn't this generate N idempotency keys out of the same string? (basically cloning a string N times?)

Reply:
It generates a unique idempotency key for each iteration, based on the same node ID.